Source/santad/DataLayer/WatchItems.h

```diff
@@ -67,16 +67,47 @@ struct WatchItemsState {
   NSTimeInterval last_config_load_epoch;
 };
 
+class DataWatchItems {
+ public:
+  DataWatchItems()
+      : tree_(std::make_unique<santa::PrefixTree<std::shared_ptr<DataWatchItemPolicy>>>()) {}
+
+  DataWatchItems(DataWatchItems &&other) = default;
+  DataWatchItems &operator=(DataWatchItems &&rhs) = default;
+
+  // Copying not supported
+  DataWatchItems(const DataWatchItems &other) = delete;
+  DataWatchItems &operator=(const DataWatchItems &other) = delete;
+
+  bool operator==(const DataWatchItems &other) const { return paths_ == other.paths_; }
+  bool operator!=(const DataWatchItems &other) const { return !(*this == other); }
+  std::vector<std::pair<std::string, WatchItemPathType>> operator-(
+      const DataWatchItems &other) const;
+
+  friend void swap(DataWatchItems &first, DataWatchItems &second) {
+    std::swap(first.tree_, second.tree_);
+    std::swap(first.paths_, second.paths_);
+  }
+
+  bool Build(std::vector<std::shared_ptr<DataWatchItemPolicy>> data_policies);
+  size_t Count() const { return paths_.size(); }
+
+  std::vector<std::optional<std::shared_ptr<DataWatchItemPolicy>>> FindPolcies(
+      const std::vector<std::string_view> &paths) const;
+
+ private:
+  std::unique_ptr<santa::PrefixTree<std::shared_ptr<DataWatchItemPolicy>>> tree_;
+  std::set<std::pair<std::string, WatchItemPathType>> paths_;
+};
+
 class WatchItems : public std::enable_shared_from_this<WatchItems> {
  public:
   using VersionAndPolicies =
-      std::pair<std::string, std::vector<std::optional<std::shared_ptr<WatchItemPolicy>>>>;
-  using WatchItemsTree = santa::PrefixTree<std::shared_ptr<WatchItemPolicy>>;
+      std::pair<std::string, std::vector<std::optional<std::shared_ptr<DataWatchItemPolicy>>>>;
```

Comment:
WDYT about adding more using decls to make this nesting a little simpler and as some of the types are used in other places? E.g. `using DataWatchItemPolicySharedPtr = std::shared_ptr<DataWatchItemPolicy` can be used on line 133 and would make the nesting here shorter. And then perhaps `using VectorOptionals = std::vector<std::optional>`?

Just a thought, feel free to ignore

Reply:
I think I over-indexed on the Google style guide of not having `using` in header files, but that isn't really the guidance. I've used some in the impl files, but I'm not consistent. For now, I'll keep as-is, but will look into simplifying these types in part 2!